receiver/loadgenreceiver/internal/list/looping.go

```diff
@@ -0,0 +1,64 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package list // import "github.com/elastic/opentelemetry-collector-components/receiver/loadgenreceiver/internal/list"
+
+import (
+	"errors"
+	"sync"
+)
+
+var ErrLoopLimitReached = errors.New("loop limit reached")
+
+// LoopingList is a list that loops over the provided list of items with an optional loop limit.
+type LoopingList[T any] struct {
+	items     []T
+	idx       int
+	loopCnt   int
+	loopLimit int
+	mu        sync.Mutex
+}
+
+// NewLoopingList returns a LoopingList over items for an optional loopLimit.
+// Setting loopLimit to 0 causes the list to loop infinitely.
```

Comment:
```suggestion
// Setting loopLimit to <= 0 causes the list to loop infinitely.
```

Reply:
You caught me putting in an undefined config value (negative range) intentionally. Let me just validate loop limit to be non-negative, so that we can define it later.